apps/services/user-profile/src/app/v2/user-profile.service.ts

```diff
@@ -114,9 +114,13 @@ export class UserProfileService {
         ].filter(Boolean),
       )
 
-      promises.map(({ confirmed, message }) => {
+      promises.map(({ confirmed, message, remainingAttempts }) => {
         if (confirmed === false) {
-          throw new BadRequestException(message)
+          throw new BadRequestException({
+            message,
+            remainingAttempts,
+            statusCode: 400,
```

Comment:
`BadRequestException` should automatically be a `400` response.

FYI this is a bit tricky to handle. I currently doesn't have any better suggestion, as this won't be used by anyone else then our profile collection UI, were we do confirmation of the email and phone number separately.

Reply:
Is this perhaps preferable?
`          throw new BadRequestException({
            message,
            statusCode: 400,
            cause: { remainingAttempts },
          })`

Reply:
@valurefugl any specific reason for wrapping the `remainingAttempts` in an object?

Reply:
Just this, https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error/cause

Reply:
As discussed offline we are not going to use the `cause` property for this.